src/arch/loongarch64.rs

```diff
@@ -0,0 +1,558 @@
+    // point just above the saved PC/RA to match the GCC/LLVM ABI.
+    "addi.d $t0, $sp, 16",
+    "st.d $t0, $a1, -16",
+    // Set up the frame pointer to point at the parent link. This is needed for
+    // the unwinding code below.
+    "move $fp, $a1",
+    // Adjust A1 to point to the parent link.
+    "addi.d $a1, $a1, -16",
+    // Pop the padding and initial PC from the coroutine stack. This also sets
+    // up the 3rd argument to the initial function to point to the object that
+    // init_stack() set up on the stack.
+    "addi.d $a2, $a2, 32",
+    // Switch to the coroutine stack.
+    "move $sp, $a2",
+    // The actual meanings of the magic bytes are:
+    // 0x0f: DW_CFA_def_cfa_expression
+    // 5: byte length of the following DWARF expression
+    // 0x86 0x70: DW_OP_breg22 (fp - 16)
+    // 0x06: DW_OP_deref
+    // 0x23, 0x10: DW_OP_plus_uconst 16
+    ".cfi_escape 0x0f, 5, 0x86, 0x70, 0x06, 0x23, 0x10",
+    // Now we can tell the unwinder how to restore the 3 registers that were
+    // pushed on the parent stack. These are described as offsets from the CFA
+    // that we just calculated.
+    ".cfi_offset 23, -16",
+    ".cfi_offset 1, -24",
+    ".cfi_offset 22, -32",
+    // As in the original x86_64 code, hand-write the call operation so that it
+    // doesn't push an entry into the CPU's return prediction stack.
+    "la.pcrel $ra, 0f",
+    "ld.d $t0, $a1, 8",
+    "jr $t0",
+    "0:",
+    asm_function_alt_entry!("stack_init_trampoline_return"),
+    // This BREAK is necessary because of our use of .cfi_signal_frame earlier.
+    "break 0",
+    ".cfi_endproc",
+    asm_function_end!("stack_init_trampoline"),
+);
+
+global_asm!(
+    // See stack_init_trampoline for an explanation of the assembler directives
+    // used here.
+    ".balign 4",
+    asm_function_begin!("stack_call_trampoline"),
+    ".cfi_startproc",
+    cfi_signal_frame!(),
+    // At this point our register state contains the following:
+    // - SP points to the top of the parent stack.
+    // - FP holds its value from the parent context.
+    // - A2 is the function that should be called.
+    // - A1 points to the top of our stack.
+    // - A0 contains the argument to be passed to the function.
+    //
+    // Create a stack frame and point the frame pointer at it.
+    "addi.d $sp, $sp, -16",
+    ".cfi_def_cfa_offset 16",
+    "st.d $ra, $sp, 8",
+    "st.d $fp, $sp, 0",
+    "addi.d $fp, $sp, 16",
+    ".cfi_def_cfa 22, 0",
+    ".cfi_offset 1, -8",
+    ".cfi_offset 22, -16",
+    // Switch to the new stack.
+    "move $sp, $a1",
+    // Call the function pointer. The argument is already in the correct
+    // register for the function.
+    "jirl $ra, $a2, 0",
+    // Switch back to the original stack by restoring from the frame pointer,
+    // then return.
+    "addi.d $sp, $fp, -16",
+    "ld.d $ra, $sp, 8",
+    "ld.d $fp, $sp, 0",
+    "addi.d $sp, $sp, 16",
+    "jr $ra",
```

Comment:
A minor change that don't use pseudo instruction to compatible with old assembler.